sql/sql_repl.cc

```diff
@@ -2232,15 +2239,118 @@ send_event_to_slave(binlog_send_info *info, Log_event_type event_type,
   }
 
   /*
-    Skip events with the @@skip_replication flag set, if slave requested
-    skipping of such events.
+    Skip events that have either
+      1) @@skip_replication flag
+      2) Set during binlogging (set using binlog_dump_* filter) 
```

Comment:
Isn't the `@@skip_replication` flag set during binlogging? So the 2nd point is really just the 1st. I'd reword as something like

> Skip events with the @@skip_replication flag set, i.e. either if slave requested skipping of such events, or if filtered via the binlog_dump_* filter options.